src/extension.ts

```diff
@@ -53,6 +54,10 @@ function didOpenTextDocument(document: TextDocument, context: ExtensionContext):
         return;
     }
     folder = getOuterMostWorkspaceFolder(folder);
+    folder = getCargoTomlWorkspace(folder, document.uri.fsPath);
+    if (!folder) {
+        return;
```

Comment:
Can/should we issue a warning about missing Cargo.toml here?

Reply:
yes probably a good idea

Reply:
My only concern is that you will get a pop up if just opening a .rs file outside it's project

Reply:
Okay I've come up with a solution that will work in both use cases.